### PR TITLE
Ensure changing arrays are notified.

### LINKED ISSUE #467: model.reload() don't update fragmentArray properties (using v6+ and ember-data 4.6)
Hello!

First thanks a lot for the awesome work to make this repository compatible with Ember 4+ 🙏 

With version 6 of this addon and ember-data 4.6 we found a little issue, when using `model.reload()`, `fragmentArray` properties are not updated correctly

https://github.com/adopted-ember-addons/ember-data-model-fragments/assets/56396753/f18349d0-72d3-4e20-9520-1889b74784cf

If we are looking at `model.currentState.recordData.__fragmentData['myProperty']`, it contains correct values, but `model['myProperty']` still contains the initial value

Reproduced here -> https://github.com/ndekeister-us/ember-test-app_ember-data-model-fragments

### PR DESCRIPTION
**Pull request by deanmarano**
I identified [this code as problematic](https://github.com/adopted-ember-addons/ember-data-model-fragments/issues/467#issuecomment-1624297238) previously, but the tests passed without it so [a partial fix was merged](https://github.com/adopted-ember-addons/ember-data-model-fragments/pull/469). More tests are needed! This would fix #467.

### CODE REVIEW

File: addon/record-data.js [hunk 1]
```diff
@@ -692,7 +692,14 @@ export default class FragmentRecordData extends RecordData {
       if (this._fragments[key]) {
         continue;
       }
-      if ((updates[key] === null) !== (original[key] === null)) {
+      const wasOrBecomingNull = original[key] === null || updates[key] === null;
+      let isChangedArray = original[key]?.length || updates[key]?.length;
+      if (isChangedArray) {
+        isChangedArray = !updates[key]?.filter?.(
+          (record) => !original[key]?.includes(record.clientId)
+        );
```
Comment (knownasilya) [inline addon/record-data.js:700]: Could the updates have an entry for a new value? In that case this wouldn't be marked as changed, right?

**Review by knownasilya — CHANGES_REQUESTED** [review]

File: addon/record-data.js (hunk 1 above)
Comment (deanmarano) [reply addon/record-data.js:700]: This first attempt has some issues. Looking at this a bit more, I see we use `diffArray` in `addons/array/stateful.js`. Maybe we should use that here as well?

Comment (knownasilya) [reply addon/record-data.js:700]: Unfamiliar with it, but all for reuse.

Comment (deanmarano) [reply addon/record-data.js]: Used that and added a test as well. Tested it with the example in #467 and it worked there as well.

**Review by knownasilya — APPROVED** [review]
❤️

**Pull request merged**